spharpy/samplings/coordinates.py

```diff
@@ -310,3 +310,19 @@ def n_max(self, value):
             self._n_max = None
         else:
             self._n_max = int(value)
+
+    def _set_points(self, x, y, z):
+        # set points first to make sure data is valid
+        super()._set_points(x, y, z)
+
+        # check if all points have the same radius
+        radius = self.radius
+        radius_delta = np.max(np.abs(np.mean(radius) - radius))
+        # add a tolerance of 1 mm
+        radius_tolerance = .001
```

Comment:
This could be simplified using np.unique(radius) or np.unique(np.round(radius, desired_precision))
https://numpy.org/doc/stable/reference/generated/numpy.unique.html

Reply:
np.round would only work if we use a decimal precision. But since we used tolerance values in other parts of pyfar, I chose to implement it this way.